Fix sigmoid operator to support boolean tensor inputs

notkisk opened the pull request:
This PR fixes the issue where boolean tensors were rejected by the `sigmoid` operator in ExecuTorch.
Specifically, it removes the rejection check for boolean tensors in `op_sigmoid.cpp` and adds proper conversion logic:

* `true` → `1.0` → `sigmoid(1.0) ≈ 0.731059`
* `false` → `0.0` → `sigmoid(0.0) = 0.5`

This resolves the failure reported in **#13492**, where a boolean tensor with shape `(4, 7, 1, 1, 7, 2)` could not be processed by `sigmoid.default`.

### Changes

* Removed boolean rejection check in `op_sigmoid.cpp`.
* Added boolean-to-float conversion logic (`true -> 1.0`, `false -> 0.0`) before applying sigmoid.
* Added comprehensive boolean tensor tests in `op_sigmoid_test.cpp`.

### Fixes

Fixes #13492 

### Test Plan

* Added new unit tests in `op_sigmoid_test.cpp` to validate behavior with boolean tensors.
* Verified that boolean tensors now produce correct sigmoid outputs without rejection.

notkisk committed:
Fix sigmoid operator to support boolean tensor inputs

- Remove boolean rejection check in op_sigmoid.cpp
- Add boolean tensor conversion logic: true->1.0, false->0.0
- Add comprehensive boolean tensor tests in op_sigmoid_test.cpp
- Fixes ExecuTorch execution failed for sigmoid.default #13492

The fix converts boolean tensors to float before applying sigmoid:
- Boolean true -> 1.0 -> sigmoid(1.0) ≈ 0.731059
- Boolean false -> 0.0 -> sigmoid(0.0) = 0.5

This resolves the original FACTO issue where boolean tensors with
shape (4, 7, 1, 1, 7, 2) were rejected by the sigmoid operator.

swolchok reviewed (changes requested):
This PR appears to add trailing whitespace. Can you please fix that so it's easier to see what lines changed?

swolchok commented on kernels/test/op_sigmoid_test.cpp:
```diff
@@ -50,6 +50,38 @@ class OpSigmoidOutTest : public OperatorTest {
+    // Destination for the sigmoid operator.
+    Tensor out = tf_out.zeros(sizes);
+
+    // Test with boolean tensor: [True, False, True, False]
+    // True should convert to 1.0 and produce sigmoid(1.0) ≈ 0.731059
+    // False should convert to 0.0 and produce sigmoid(0.0) = 0.5
```
please remove comments like this one that repeat the code

notkisk committed:
Fix sigmoid operator to support boolean tensor inputs

This PR fixes the issue where boolean tensors were rejected by the sigmoid operator in ExecuTorch.
Specifically, it removes the rejection check for boolean tensors in op_sigmoid.cpp and adds proper conversion logic:

true → 1.0 → sigmoid(1.0) ≈ 0.731059
false → 0.0 → sigmoid(0.0) = 0.5

This resolves the failure reported in #13492, where a boolean tensor with shape (4, 7, 1, 1, 7, 2) could not be processed by sigmoid.default.

Changes
- Removed boolean rejection check in op_sigmoid.cpp
- Added comprehensive boolean tensor tests in op_sigmoid_test.cpp

Fixes #13492

Test Plan
Added new unit tests in op_sigmoid_test.cpp to validate behavior with boolean tensors.
Verified that boolean tensors now produce correct sigmoid outputs without rejection.

swolchok commented:
can you please rebase and run `lintrunner init && lintrunner -a`?

swolchok commented:
the lintrunner message in CI is formatted a bit oddly, but I repro lintrunner not being clean:

```
~/src/executorch> lintrunner
Warning: Could not find a lintrunner config at: '.lintrunner.private.toml'. Continuing without using configuration file.
  FLAKE8 success!
  UFMT success!
  CLANGFORMAT failure
  CMAKEFORMAT success!
  CMAKE success!
  ETCAPITAL success!
  NEWLINE success!
  NOSTDINC success!
  NOTORCHINC success!
  MYPY success!
  LICENSELINT success!
  TORCH_AO_IMPORT success!


>>> Lint for kernels/test/op_sigmoid_test.cpp:

  Warning (CLANGFORMAT) format
    See https://clang.llvm.org/docs/ClangFormat.html.
    Run `lintrunner -a` to apply this patch.

    You can run `lintrunner -a` to apply this patch.

    62  62 |     op_sigmoid_out(tf.make(sizes, /*data=*/{true, false, true, false}), out);
    63  63 |
    64  64 |     EXPECT_TENSOR_CLOSE(
    64     |-        out,
    65     |-        tf_out.make(sizes, /*data=*/{0.731059, 0.5, 0.731059, 0.5}));
        65 |+        out, tf_out.make(sizes, /*data=*/{0.731059, 0.5, 0.731059, 0.5}));
    67  66 |
    68  67 |     out = tf_out.zeros({3});
    69  68 |     op_sigmoid_out(tf.make({3}, /*data=*/{true, true, true}), out);
```

notkisk committed:
chore(lint): apply clang-format fixes from lintrunner

swolchok reviewed (approved):
OK, the CI results look sufficient to land. Apparently I need to accept because I previously requested changes, so doing that.
